### PR TITLE
A couple of refactoring / tweak

### PR DESCRIPTION
**Pull request by tninja**
- Break the large function into smaller ones
- Make it a little bit more functional, eg. for rewrite, make it possible to have shortcut function of instruction
- Show the chatgpt answer by switching to *ancilla-chat* buffer

(Rewrite / refactor mostly by ancilla.el itself)

@shouya PTAL Thanks

### CODE REVIEW

File: ancilla.el [hunk 1]
```diff
@@ -274,11 +293,11 @@ Return a relative path when a project is detected."
...
-                   ((and (fboundp 'ffip-project-root)
-                         (ffip-project-root)))
-                   ((and (fboundp 'project-current)
-                         (fboundp 'project-root)
-                         (project-root (project-current))))
+                   ;; ((and (fboundp 'ffip-project-root)
+                   ;;       (ffip-project-root)))
+                   ;; ((and (fboundp 'project-current)
+                   ;;       (fboundp 'project-root)
+                   ;;       (project-root (project-current))))
```
Comment (tninja) [inline ancilla.el:300]: this is just fail on my laptop. so i comment them out

File: ancilla.el [hunk 2]
```diff
@@ -74,6 +74,12 @@ Possible values are t (always), 'rewrite-only, 'generate-only, or nil (never)."
                  (const :tag "Never" nil))
   :group 'ancilla)
 
+(defcustom ancilla-chat
```
Comment (tninja) [inline ancilla.el:77]: introduce a const variable

File: ancilla.el [hunk 3]
```diff
@@ -106,57 +112,70 @@ call 'ancilla-rewrite' otherwise."
...
 
+(defun ancilla--process-ask (instruction)
+  "Process the given INSTRUCTION for 'ask' mode."
+  (ancilla--process-instruction instruction 'ask))
+
+(defun ancilla--process-generate (instruction)
+  "Process the given INSTRUCTION for 'generate' mode."
+  (ancilla--process-instruction instruction 'generate))
+
+(defun ancilla--process-rewrite (instruction)
```
Comment (tninja) [inline ancilla.el:137]: this function can provide a shortcut for given fixed rewrite instruction (maybe it is possible)

File: ancilla.el [hunk 4]
```diff
@@ -499,7 +517,12 @@ replacement text as argument."
    (concat "User selected the region marked by <|begin selection|>/<|end selection|> and asked:\n"
            instruction))
 
-  (ancilla--adaptor-chat-request-buffer-send 'message))
+  (ancilla--adaptor-chat-request-buffer-send (lambda (message)
+                                               ())) ;; discard the message, we can print to some buffer in future if we want
+
+  ;; show result
+  (switch-to-buffer-other-window ancilla-chat)
```
Comment (tninja) [inline ancilla.el:524]: since the answer could be long, personally feeling switch to ancilla-chat window will be much easier to read

**Comment by shouya** [comment]
Thank you for the contribution!

**Pull request merged**